### PR TITLE
feat: [PIE-7445]: Added support for webhookId for github webhook trigger

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots
Screenshot 2023-01-16 at 10 15 29 AM

Screenshot 2023-01-16 at 10 13 51 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger cypress-pipeline

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger cypress-pipeline

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx [hunk 1]
```diff
@@ -341,7 +341,8 @@ export default function WebhookTriggerWizard(
       autoAbortPreviousExecutions: false,
       pipelineBranchName: getDefaultPipelineReferenceBranch(baseType) || branch,
       // setDefaultValue only when polling is enabled and for Github Webhook Trigger
-      ...(isGitWebhookPollingEnabled && sourceRepo === GitSourceProviders.GITHUB.value && { pollInterval: '0' })
+      ...(isGitWebhookPollingEnabled &&
+        sourceRepo === GitSourceProviders.GITHUB.value && { pollInterval: '0', webhookId: '' })
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx:345]: Should this be behind the git polling FF?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx:345]: Yes. This should comes only with git polling field. Webhook Id will be used when there is issue in automatic trigger registration

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/72-triggers/pages/triggers/utils/WebhookTriggerConfigPanelUtils.tsx [hunk 2]
```diff
@@ -155,19 +155,27 @@ export const renderNonCustomEventFields = ({
...
-                dataTooltipId: 'pollInterval'
-              }}
-              multiTypeDurationProps={{
-                enableConfigureOptions: false,
-                allowableTypes: [MultiTypeInputType.FIXED],
-                width: 324
-              }}
-              placeholder={getString('triggers.triggerConfigurationPanel.pollingFrequencyPlaceholder')}
-            />
+            <>
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/pages/triggers/utils/WebhookTriggerConfigPanelUtils.tsx:158]: Looks like duplicated code. Is it possible to take common code into once component and reuse it as both the places ?

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

File: src/modules/72-triggers/pages/triggers/utils/WebhookTriggerConfigPanelUtils.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/72-triggers/pages/triggers/utils/WebhookTriggerConfigPanelUtils.tsx:158]: @chetanvekariya-harness this is due to refactoring. Code from one place will go away once we remove `CD_TRIGGERS_REFACTOR` FF.

**Pull request merged**